### PR TITLE
Add claim vesting iframe hook

### PR DESCRIPTION
**Pull request by JeanNeiverth**
# Summary

- Create the Claim Vesting Hook in the dapps folder as an IFrame linked to Bleu vercel deployed app
- Add Claim Vesting Hook in the hookStore, as both pre and post hook, in all chains the LlamaPay vesting factory is implemented (Mainnet, Gnosis and Arbitrum)
- Edit IFrameContainer min-height to 300px for aesthetic reasons

image

image

image

image

Errors cases:
- Insert a valid Ethereum address
- Please connect your wallet first
- Address is not a valid vesting contract
- You are not the contract recipient

Example:
image

Success case:
image

# How to test:

1. Access /#/100/swap/hooks/WXDAI

2. Connect your wallet and switch to Gnosis chain

3. In Hooks > Add Pre-Hook Action, the Claim Vesting Hook should appear in the "All hooks" section

4. Insert a valid vesting contract. Here is a mapping of addresses to vesting contracts that should work:

Address:
0x9FA3c00a92Ec5f96B1Ad2527ab41B3932EFEDa58
Vesting Contract:
0x77e62665422fc216ca278b7655b3329dc5ba173c

Address:
0xFF714b8b0e2700303eC912BD40496C3997ceEa2b
Vesting Contract:
0xf078d4ADBa99BaC7Ab4a64ea3F24b6307107198d

Address:
0x8FAb71C0d4272698A3B2d1F3Ed5FC3c1B9b3E531
Vesting Contract:
0xac7CBD5DD1D64bF6C9cdCceCf0cb7c909C3AF4A6

### CODE REVIEW

**Comment by JeanNeiverth** [comment]
Just letting a note here: if you change the CoW light/dark mode during the hook app execution, you will see it does not change the hook theme, although it will set the theme correctly if you quit and reopen the hook.

The hook is implemented in a way it should respond to context.isDarkMode changes, so I think what is causing the issue is the @cowprotocol/hook-dapp-lib needs to be updated to trigger this theme change, because currently typescript can't even recognize the context.isDarkMode parameter, I had to ts-ignore it to be able to deploy.

@alexdorsch @alfetopito can you check this?

**Comment by elena-zh** [comment]
Hey @JeanNeiverth , great job!  Thank you for building this hook, and for the great PR description. 

Still, I have some issues found: 
1. General styling: I think, all elements should look more or less the same across all available hooks in the app. So, it would be great to:
- change 'Add hook' button to 'Add Pre-hook' and 'Add Post-hook' (accordingly) like we do for 'Claim GNO from validators' and other hooks
post-pre
- same for 'edit hook' button: It would be nice to name it 'Update Pre-hook' and 'Update Post-hook' (accordingly)
- hide the 'available amount' field to a not connected user
hide
- hide the 'available amount' field when it is 0
- show a validation message in the same manner as we do for other hooks. Examples: 
example
validation message

- truncate trailing and preceding spaces in the address field
spaces

2. There are also some style issues like: 
- it would be great to improve the 'Total Available' field in a mobile view
mobile view
- I think, it would be nice to remove 'Ethereum' word from the 'Insert a valid Ethereum address'  validation message as it does not conform to Gnosis chain
ethereum - generic

- Fix an issue with the fields color on a network change: when this hook details is open, switch the there from light to dark (or vice versa): 
fields

3. Hooks execution/validation: 
- Is this OK that I have some dust left to claim after a hook execution? 
dust
- Is this OK to see available to claim amount for another wallet?
- When I'm connected to an unsupported chain, I'm still able to add a hook. Is this OK for now? 
chain
- I see different validation messages for the same vesting address on Ethereum and Arbitrum. Could you please clarify why? 
arb
ethereum

4. (I'd be very thankful): could you please deploy a vesting contract for SC this `0x349C217d5E4F41a1d1C9a2189d040110dea93659 `address on Gnosis chain so I can test it there?

**Comment by JeanNeiverth** [comment]
Hi @elena-zh, thanks for the review!

I could handle most of the points, but others I believe will need other actions by the CoW team side

## Fixed issues:

- change 'Add hook' button to 'Add Pre-hook' and 'Add Post-hook' (same for edit)
- change "Edit" to "Update"
- hide the 'available amount' field to a not connected user
- hide the 'available amount' field when it is 0
- add clean UI with "connect your wallet" message when wallet isn't connected
- general errors passed to inside the main button
- address input now doesn't accept spaces / any invalid character
- add shorter message and formatting in "Claimable amount" output, turning it better to mobile users
- remove "Ethereum" in the invalid 'Ethereum' address error
- **Is this OK that I have some dust left to claim after a hook execution?** Yep, it's the expected behavior of the vesting contract, as it releases money with time
- **I see different validation messages for the same vesting address on Ethereum and Arbitrum. Could you please clarify why?** Sure! Each network is a different environment, so a contract created in an Arbitrum address will not appear in the Ethereum network, unless the same contract is created there too. In this case, probably the address in the Ethereum net is an empty one.
- Deployed a contract in Gnosis for `0x349C217d5E4F41a1d1C9a2189d040110dea93659`. Contract address is `0x349C217d5E4F41a1d1C9a2189d040110dea93659`

Note: as this hook is an IFrame, you no longer have to wait a new deploy on your side (as it is deployed in Bleu side), so you can already check these changes in the same app you ran before. The only changes that will require a deploy on CoW side are the ones below.

## Pending

- Dark / Light mode switching while executing hook
- **Is this OK to see available to claim amount for another wallet?** I can't see any problem, since the data is already public. But we can ping @alfetopito and @alexdorsch to confirm that
- **When I'm connected to an unsupported chain, I'm still able to add a hook. Is this OK for now?** This is not expected, but we are working on a solution

When these pending topics get resolved, I will ping you again here. Feel free to add other comments / issues in the mean time if you like :)

**Comment by elena-zh** [comment]
Hey @JeanNeiverth , thank you! 
Almost all issues were fixed. I've been able to find some nitpicks only: 
1. It would be nice to show a full amount in a title when hover a mouse on a claimable amount (like we do everywhere in the app): 
hover
2. (uber-nitpick) but I see that fonts in the 'Claim GNO' and 'Claim vesting' hooks differ from each other a bit. Could you please double check this?
fonts diff
3. Could you please share with me a vesting contract link for my Safe address? 

> Deployed a contract in Gnosis for 0x349C217d5E4F41a1d1C9a2189d040110dea93659. Contract address is 0x349C217d5E4F41a1d1C9a2189d040110dea93659

Thanks!

**Comment by JeanNeiverth** [comment]
Hi @elena-zh, thank you again for the careful review!

By now all the issues were resolved, including:
- Dark / Light mode switching while executing hook now is working
- Not showing claimable amount for wallets different than recipient anymore
- Changing to an unsupported chain now closes the hook app

Note that for these changes you will need a deploy on CoW side @alexdorsch @alfetopito 

About the most recent comments:
1. Done! Please, could you check again if this feature is implemented as expected?
2. The only difference I could find was an additional opacity of 80% when disabled in the new button, that I removed. Font size, weight and color are the same. Could you verify again if the difference is still there?
3. Sorry about last message, I just copied the wrong address... the right one is here: `0x3bc7ef9924ddbb0e836d01a79e0fa59423da6210`

Thanks for your patience / I apologize for the response delay and please let me know if there are other changes you'd like to see

**Comment by elena-zh** [comment]
@JeanNeiverth , thank you!

Please don't kill me, but I've found some more issues to report: 
1. There is no UI validation if I've already claimed my finds. It would be great to ad it:
have already claimed
2. As for the button's font properties, I've also tried to find where a discrepancy is. Most likely it is in here: when I set a font-family instead of 'inherit' property, for me it starts looking a bit better
imherit
3. I've noticed that now 'disabled' button state is missing: was this done intentionally? 
disabled state
like here: 
disabled state example
4. The last nitpick: it would be great if these fields will have the same bg color: 
different colors
same color

like it is everywhere in the app. See examples: 
image
image


Thanks!

**Comment by JeanNeiverth** [comment]
@elena-zh, no worries!

Just fixed these points, you can already see the changes if you access the previous deploy.

Thanks again and do not hesitate pointing more issues if you think its necessary :)

**Comment by alfetopito** [comment]
I'm been slow to see GH notifications of late.
BTW, Sasha's GH handle is @shoom3301.

File: libs/hook-dapp-lib/src/hookDappsRegistry.json [hunk 1]
```diff
@@ -49,5 +49,19 @@
     "conditions": {
       "supportedNetworks": [11155111]
     }
+  },
+  "CLAIM_VESTING": {
+    "id": "177eda1e6d1d89f1777d60387f06e1fdcb2c11baad38c6f06c8a3404ebc7e03a",
+    "type": "IFRAME",
+    "name": "Claim Vesting Hook",
+    "description": "The Claim Vesting Hook is a powerful and user-friendly feature designed to streamline the process of claiming funds from LlamaPay vesting contracts. This tool empowers users to easily access and manage their vested tokens, ensuring a smooth and efficient experience in handling time-locked assets.",
+    "descriptionShort": "Claim your LlamaPay vesting contract funds",
```
Comment (alfetopito) [inline libs/hook-dapp-lib/src/hookDappsRegistry.json:58]: Is this exclusive to LlamaPay?

If that's the case, I suggest to name it so to be explicit and not set wrong expectations.

**Comment by elena-zh** [comment]
Hey @JeanNeiverth , great job, thank you!

The last nitpick is that it would be great to reduce a font-size on a button in a mobile view
image
see examples
image
image

**Comment by JeanNeiverth** [comment]
Hi guys!

Just to say I fixed these last requests @elena-zh @alfetopito. Sorry for missing @shoom3301 GH name before, that was completely my bad, but feel free to review too.

Thanks for your patience and reviews

**Review by elena-zh — APPROVED** [review]
Thank you!

**Review by alfetopito — APPROVED** [review]

**Comment by elena-zh** [comment]
Hey @JeanNeiverth , can this be merged?

**Review by shoom3301 — APPROVED** [review]

**Pull request merged**